plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -77,10 +76,6 @@ public String toString() {
 
     @Override
     public void close() throws IOException {
-        if (reader instanceof BufferedReader) {
-            ((BufferedReader) reader).close();
-        } else if (reader instanceof InputStreamReader) {
-            ((InputStreamReader) reader).close();
-        }
+        read().close();
```

Comment:
Not sure this is correct either.
I am not seeing much benefit in having `YamlReader<T> reader;` and `YamlSource.of` pattern if it comes at the cost of properly closing sources.

Thoughts @oleg-nenashev @timja ?

Reply:
pretty ugly imo
```
try (YamlSource source = src; Reader r = source.read()) {
```

I haven't looked too closely to it but YamlSource imo shouldn't be opening anything, it should just be the reader

Reply:
Actually I think the very reason why sources weren't closed previously is the very fact that it uses a functional interface.

Reply:
I tried "cleaning" it up but it become more ugly.

I think YamlSource is actually holding the reader open. Basically leaking the reader in memory.
Perhaps best served with rewriting it so that the reader is only held open in the merge block.